SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -225,34 +228,56 @@ void USpatialReceiver::OnAuthorityChange(const Worker_AuthorityChangeOp& Op)
 
 void USpatialReceiver::HandlePlayerLifecycleAuthority(const Worker_AuthorityChangeOp& Op, APlayerController* PlayerController)
 {
+	USpatialNetConnection* Connection = Cast<USpatialNetConnection>(PlayerController->GetNetConnection());
+	bool bIsServer = NetDriver->IsServer();
+
 	// Server initializes heartbeat logic based on its authority over the position component,
 	// client does the same for heartbeat component
-	if ((NetDriver->IsServer() && Op.component_id == SpatialConstants::POSITION_COMPONENT_ID) ||
-		(!NetDriver->IsServer() && Op.component_id == SpatialConstants::HEARTBEAT_COMPONENT_ID))
+	if ((bIsServer && Op.component_id == SpatialConstants::POSITION_COMPONENT_ID) ||
+		(!bIsServer && Op.component_id == SpatialConstants::HEARTBEAT_COMPONENT_ID))
 	{
-		if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+		if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE && Connection != nullptr)
 		{
-			if (USpatialNetConnection* Connection = Cast<USpatialNetConnection>(PlayerController->GetNetConnection()))
+			if (bIsServer)
 			{
-				if (NetDriver->IsServer())
-				{
-					AuthorityPlayerControllerConnectionMap.Add(Op.entity_id, Connection);
-				}
-				Connection->InitHeartbeat(TimerManager, Op.entity_id);
+				AuthorityPlayerControllerConnectionMap.Add(Op.entity_id, Connection);
 			}
+			Connection->InitHeartbeat(TimerManager, Op.entity_id);
 		}
 		else if (Op.authority == WORKER_AUTHORITY_NOT_AUTHORITATIVE)
 		{
-			if (NetDriver->IsServer())
+			if (bIsServer)
 			{
 				AuthorityPlayerControllerConnectionMap.Remove(Op.entity_id);
 			}
-			if (USpatialNetConnection* Connection = Cast<USpatialNetConnection>(PlayerController->GetNetConnection()))
+			if (Connection != nullptr)
 			{
 				Connection->DisableHeartbeat();
 			}
 		}
 	}
+
+	// Initialise Latency Manager for server authoritative over the server ping component and
+	// for the client authoritative over the client pong component
+	else if ((bIsServer && Op.component_id == SpatialConstants::SERVER_PING_COMPONENT_ID) ||
+		(!bIsServer && Op.component_id == SpatialConstants::CLIENT_PONG_COMPONENT_ID))
+	{
+		if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE && Connection != nullptr)
+		{
+			Connection->SetupLatencyManager(Op.entity_id);
+		}
+		else if (Op.authority == WORKER_AUTHORITY_NOT_AUTHORITATIVE)
+		{
+			if (bIsServer)
+			{
+				ClientPongDelegates.Remove(Op.entity_id);
```

Comment:
Ahh it gets removed here. I think it would be better if the LatencyManager dealt with this removal.